source/coap_connection_handler.c

```diff
@@ -266,7 +266,10 @@ static internal_socket_t *int_socket_create(uint16_t listen_port, bool use_ephem
 
         // Set socket option to receive packet info
         socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_IPV6_RECVPKTINFO, &(const bool) {1}, sizeof(bool));
-
+        if (socket_interface_selection != -1) {
+            // Select socket interface if selection requested
+            socket_setsockopt(this->socket, SOCKET_IPPROTO_IPV6, SOCKET_INTERFACE_SELECT, &socket_interface_selection, sizeof(int8_t));
```

Comment:
Multicasts should normally be initially sent to one interface only by the application and network stack.

(The new core has a new socket option IPV6_MULTICAST_IF to specify that, which is standard - if not set, the existing socket option you're setting is used.)

If it's link-local, the specified interface is as far as it will go.

If it's wider-scope, and we are a multicast router, we should rely on the packet sent looping back into our router core to get it to the other interfaces. It gets forwarded just like any other wide-scope multicast packet received on that interface, according to the same forwarding rules.

I assume the existing code ends up sending site local to all interfaces, as it would always forward site-local in both directions - it's probably hard-coded. New core works more like the initial description - loop back into forwarder.

If you're not a multicast router, sending to 1 interface is sufficient and correct if there is some other multicast forwarding system that will ultimately forward to any other links. (Sending to each interface could lead to duplication if someone else forwards between them).

If you're not a multicast router, and you're attached to two disjoint links that each need the multicast, with no other forwarding, then the application would have to know to send to each interface separately.